### PR TITLE
HAL description

### PR DESCRIPTION
**Pull request by danko-miladinovic**
Added HAL description with a small diagram of the measurement process.

### CODE REVIEW

File: docs/hal.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Hardware Abstraction Layer (HAL)
+HAL is an abstraction layer for confidential computing. It is an in-memory virtual machine (VM). For Cocos, HAL is used with SEV-SNP to provide a fully encrypted VM.
```
Comment (drasko) [inline docs/hal.md:2]: Add explanation of the acronym - (Hardware Abstraction Layer)

Also explain that SVMs act as bare-metal machines (isolated from host OS), so we need to put some runtime in them in order to run the applications.

File: mkdocs.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ theme:
 
 markdown_extensions:
   - admonition
+  - attr_list
```
Comment (drasko) [inline mkdocs.yml:16]: Why `attr_list`? I was expecting `hal`, no?

**Review by drasko — CHANGES_REQUESTED** [review]

File: docs/hal.md (hunk 1 above)
Comment (danko-miladinovic) [reply docs/hal.md]: I have added a more detailed description of HAL.

File: mkdocs.yml (hunk 2 above)
Comment (danko-miladinovic) [reply mkdocs.yml:16]: I have added this markdown extension in order to use attributes when working with pictures. For example, `{ align=center }` in the text.

Comment (danko-miladinovic) [reply mkdocs.yml:16]: I have added a `hal.md` file and its menu entry (in `nav` in `mkdocs.yml`) in the previous pull request.

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**